From feb2da7b3a6b8d5962786b667847a3e781263505 Mon Sep 17 00:00:00 2001
From: PulkoMandy <pulkomandy@pulkomandy.tk>
Date: Tue, 9 Sep 2025 20:07:53 +0200
Subject: Fixes for Haiku:

- Use newer version of fs4 which pulls Rustix 1.x, that includes Haiku
  support
- Define what to do for Haiku in endian.h

diff --git a/Cargo.toml b/Cargo.toml
index 134bb08..e19b4d9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -118,7 +118,7 @@ ctrlc = { version = "3.4.5", features = ["termination"] }
 dialoguer = { version = "0.11.0", features = ["fuzzy-select"] }
 etcetera = "0.8.0"
 filetime = "0.2.25"
-fs4 = "0.12.0"
+fs4 = "0.13.1"
 git2 = "0.20.0"
 glob = "0.3.2"
 heck = "0.5.0"
diff --git a/lib/src/portable/endian.h b/lib/src/portable/endian.h
index f467ec5..5f88b5d 100644
--- a/lib/src/portable/endian.h
+++ b/lib/src/portable/endian.h
@@ -18,6 +18,7 @@
 #if defined(HAVE_ENDIAN_H) || \
     defined(__linux__) || \
     defined(__GNU__) || \
+    defined(__HAIKU__) || \
     defined(__illumos__) || \
     defined(__NetBSD__) || \
     defined(__OpenBSD__) || \
-- 
2.50.1

